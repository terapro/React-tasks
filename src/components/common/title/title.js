import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';


const Title = (props) => {
  let itemClass = classNames({
    'ttl': true,
    'ttl-red': props.ttlRed,
    'ttl-white': props.ttlWhite,
    'ttl-UC': props.ttlUC,
    'ttl-big': props.ttlBig

  });

  return (<h2 className={itemClass}>{props.content}</h2>);
};


Title.propTypes = {
  ttlBig: PropTypes.bool,
  ttlUC: PropTypes.bool,
  ttlRed: PropTypes.bool,
  ttlWhite: PropTypes.bool,
  content: PropTypes.string
};
Title.defaultProps = {
  content: 'Assign a title'
};

export {Title};