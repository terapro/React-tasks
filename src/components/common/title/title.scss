.title {
  text-transform: uppercase;
  color: #FFF;
  font-size: 22px;
}