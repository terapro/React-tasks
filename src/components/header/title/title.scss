.header-title {
  margin-top: 20px;
}