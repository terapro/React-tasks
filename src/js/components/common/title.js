import React from 'react';
import PropTypes from 'prop-types';

const Title = (props) => (
    <h2 className={'title'}>{props.content}</h2>

);

Title.propTypes = {
    content: PropTypes.string
};

export {Title};