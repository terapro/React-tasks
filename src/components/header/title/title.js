import React from 'react';

import {Title} from '../common/title/title.js'

const Title = () => (
    <div className={'header-title'}>
    <Title content={"find your movie"} />
    </div>

);

export {Title};