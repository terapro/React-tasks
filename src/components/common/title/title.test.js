import React from 'react';
import {Title} from 'src/components/common/title/title.js';
import {shallow} from 'enzyme';

const mockContent = 'Testing title';

describe('<Title />', () => {
  describe('Rendering', () => {
    it('should render the component', () => {
      const wrapper = shallow(<Title content={mockContent} />);
      expect(wrapper).toMatchSnapshot();
    });
    it('should render the component with default content', () => {
      const wrapper = shallow(<Title content={mockContent} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('Configuration', () => {
    it('can set the white color of the text', () => {
      expect(shallow(<Title ttlWhite />).is('.ttl-white')).toBe(true);
    });
    it('can set the red color of the text', () => {
      expect(shallow(<Title ttlRed />).is('.ttl-red')).toBe(true);
    });
    it('can set the upper case for the text', () => {
      expect(shallow(<Title ttlUC />).is('.ttl-UC')).toBe(true);
    });
    it('can set big size for the text', () => {
      expect(shallow(<Title ttlBig />).is('.ttl-big')).toBe(true);
    });
  });
});
