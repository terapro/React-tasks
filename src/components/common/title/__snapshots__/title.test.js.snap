// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> Rendering should render the component 1`] = `
<h2
  className="ttl"
>
  Testing title
</h2>
`;

exports[`<Title /> Rendering should render the component with default content 1`] = `
<h2
  className="ttl"
>
  Testing title
</h2>
`;

exports[`<Title /> should render the title 1`] = `
<h2
  className="ttl"
>
  Testing title
</h2>
`;
